doc/tutorials/enable_s5.rst

```diff
@@ -0,0 +1,364 @@
+     :caption: RT-Linux launch script
+  
+     # for pm setting
+     pm_channel="--pm_notify_channel uart "
+     pm_by_vuart="--pm_by_vuart tty,/dev/ttyS1"
+     
+     /usr/bin/acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+        --lapic_pt \
+        --rtvm \
+        --virtio_poll 1000000 \
+        -U 495ae2e5-2603-4d64-af76-d4bc5a8ec0e5 \
+        -s 2,passthru,02/0/0 \
+        -s 3,virtio-console,@stdio:stdio_port \
+        $pm_channel $pm_by_vuart \
+        --ovmf /usr/share/acrn/bios/OVMF.fd \
+        hard_rtvm
+     
+     }
+
+  .. note:: For RT-Linux, the vUART is emulated in hypervisor, expose the node as ``/dev/ttySn``.
+
+#. For LaaG and RT-Linux VM, a reference daemon is offered to run:
+
+   a. Use below command lines to make ``life_mngr``.
+
+      .. code-block:: none
+   
+         $ cd acrn-hypervisor/misc/life_mngr
+         $ gcc life_mngr.c -o life_mngr
+   
+      .. note:: You can also use ``make life_mngr`` command under ``acrn-hypervisor/misc`` directory,
+         it also will build ``life_mngr``.
+
+   #. Copy ``life_mngr`` and ``life_mngr.service`` into User VM:
+      ``/usr/bin/life_mngr``, and ``/lib/systemd/system/life_mngr.service``
+
+   #. Use below commands to enable ``life_mngr.service`` and restart User VM.
+
+      .. code-block:: none
+
+         # chmod +x /usr/bin/life_mngr
+         # chmod +x /lib/systemd/system/life_mngr.service
+         # systemctl enable life_mngr.service
+         # reboot
+
+#. For WaaG VM, a reference daemon is offered to run:
+
+   a) Build ``life_mngr_win.exe`` application:
+
+      I) Go to ``acrn-hypervisor/misc/life_mngr``, then run
+         ``x86_64-w64-mingw32-gcc -g life_mngr_win.c -o life_mngr_win.exe`` under docker environment.
+
+      #) Or you can also enter ``acrn-hypervisor/misc`` directory and run ``make life_mngr``
+         to build ``life_mngr_win.exe``.
+
+      #) If there is no ``x86_64-w64-mingw32-gcc`` compiler, you need run ``swupd bundle-add c-basic-mingw``
+         to install it.
+
+   #) Set up a Windows environment:
+
+      I) Download the :kbd:`Visual Studio 2019` tool from `<https://visualstudio.microsoft.com/downloads/>`_,
+         and choose the two options in the below screenshots to install "Microsoft Visual C++ Redistributable
+         for Visual Studio 2015, 2017 and 2019 (x86 or X64)" in WaaG:
+
+         .. figure:: images/Microsoft-Visual-C-install-option-1.png
+
+         .. figure:: images/Microsoft-Visual-C-install-option-2.png
+
+      #) In WaaG, use the :kbd:`WIN + R` shortcut key, input "shell:startup", click :kbd:`OK`
+         and then copy ``life_mngr_win.exe`` application into this directory.
+
+         .. figure:: images/run-shell-startup.png
+
+         .. figure:: images/launch-startup.png
+
+   #) Restart the WaaG VM. The COM2 window will automatically open after reboots.
+
+         .. figure:: images/open-com-success.png
+
+#. If the Service VM is being shut down (transitioning to the S5 state), it can call
```

Comment:
> Can you add a note to indicate that this is not automatically triggered by a Service VM shutdown, this needs to be run before powering off the Service VM.

Note is added.

> Would it be possible to make this automatic in the Service VM by the way (in a future iteration of the tool)?
@mcao6 , could you please help on this ?

Reply:
Yes, it can be. But as discussion with Fengwei before, actually it depends more on  the specific hardware target: if S5 triggered by IOC in Service VM, we need integrated ours into it; if it triggered by a power button which pass-through to Service VM, we need do some modification in system "shutdown service", even the "shutdown" program, it is hard to maintain in our ACRN project. So current we offer a reference solution first, as product, the work can be done further.